messaging/firebase-messaging-sw.js

```diff
@@ -19,9 +19,12 @@ const messaging = firebase.messaging();
  importScripts('https://www.gstatic.com/firebasejs/7.9.1/firebase-messaging.js');
 
  // Initialize the Firebase app in the service worker by passing in the
- // messagingSenderId.
+ // apiKey, projectId, messagingSenderId and appId values.
  firebase.initializeApp({
-   'messagingSenderId': 'YOUR-SENDER-ID'
+   'apiKey': 'YOUR-API-KEY',
+   'projectId': 'YOUR-PROJECT-ID',
+   'messagingSenderId': 'YOUR-SENDER-ID',
+   'appId': 'YOUR-APP-ID'
```

Comment:
Another option might just be to tell developers to insert their entire Firebase config object. cc @kroikie

Reply:
Chatted with Arthur. I'm going to suggest a change

Reply:
> We could also modify the code to throw an error if a proper config isn't passed in.

Good idea. I have a couple of comments on it.

1. Is there a need to do this? All previous versions of docs (I guess) didn't have `apiKey`, `projectId` and other configurations specified and it worked just fine with the users.

2. If we do this, maybe we can add a comment over that `if` block specifying to remove the block after the Firebase config object is placed. Something like the following. This makes it clear that the block is a temporary code and can/should be removed.

```js
// [START_EXCLUDE]
// Remove the following block after updating the Firebase config object above
```

Reply:
> We could also modify the code to throw an error if a proper config isn't passed in.

Good idea. I have a couple of comments on it.

1. Is there a need to do this? All previous versions of docs (I guess) didn't have `apiKey`, `projectId` and other configurations specified and it worked just fine with the users.

2. If we do this, maybe we can add a comment over that `if` block specifying to remove the block after the Firebase config object is placed. Something like the following. This makes it clear that the block is a temporary code and can/should be removed.

```js
// [START_EXCLUDE]
// Remove the following block after updating the Firebase config object above
```

Reply:
1. It isn't necessary to include the entire config object, but the thought is "why not" - it seems easier to copy and paste the whole thing from the Firebase console (a [well-documented process](https://firebase.google.com/docs/web/setup#config-object)) than to pick out the `apiKey`, `projectId`, `messagingSenderId` and `appId` fields.
2. Sure! Sounds good to me 😀

Reply:
> It isn't necessary to include the entire config object, but the thought is "why not"

@jhuleatt Ah, my `#1` comment was also for the second part of your suggestion which involves checking `appId` for a non-default value. Since we haven't been doing this so far (providing a check for `appId` or firebase object), do users need this?

Reply:
Oh, I see. Yeah it isn't totally necessary, and there's a chance the code/confusion it adds might outweigh the benefit of a slightly nicer error message. It's up to you - I'm happy to merge with or without it.

Reply:
Actually, since you've got it all set with the new config object, I'm just going to merge as-is. We can always work on that check in another PR.

Reply:
Thanks for merging. 

> ... there's a chance the code/confusion it adds might outweigh the benefit of a slightly nicer error message. It's up to you - I'm happy to merge with or without it.

I think we shouldn't add it because most developers are already aware they need to put their Firebase config object (or something to authenticate their firebase app) somewhere for them to use a Firebase app. So it's unlikely they will do this mistake.

Also, the code file isn't too long for someone to miss the part that involves setting the firebase config object.